app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -135,11 +136,25 @@ const selectableRowLimits = computed(() => {
 })
 const wasAutomaticallyAutosized = ref(false)
 
-const numberFormat = new Intl.NumberFormat(undefined, {
-  style: 'decimal',
-  maximumFractionDigits: 12,
-  useGrouping: 'min2' as any,
-})
+function isGrouingRequired(params: ICellRendererParams) {
+  const field = params.colDef?.field
+  const allDataForCol = params.node.parent?.allLeafChildren.map((row) => {
+    const data = row.data[field || '']
+    return data?.type === 'BigInt' ? data.value : data
+  })
+  return allDataForCol?.find((num) => num > 9999) != null
+}
+
+function formatNumber(params: ICellRendererParams) {
+  const isGrouingRequiredForData = isGrouingRequired(params)
+  const valueType = params.value?.type
+  const value = valueType === 'BigInt' ? BigInt(params.value?.value) : params.value
+  return new Intl.NumberFormat(undefined, {
```

Comment:
Let's make two `Intl.NumberFormat` instances and reuse them across `formatNumber` calls. While nowadays it's not as important as it used to be thanks to many browser optimizations, it is still recommended to cache those for performance reasons. Creating the formatter instance can surprisingly have a significant cost.

Reply:
Thank you!